### DISCUSSION TITLE
SolidHfus

### DISCUSSION THREAD

**Discussion by Kelethabetse** [ClapeyronThermo/Clapeyron.jl#200] [General]
Dear @pw0908 and @longemen3000 , thank you for the new solid solubility update! It is helpful a great deal❤️

I would kindly like to ask, the solids I am sampling aren't in the file on Clapeyron hence when I create a model I get an error message regarding missing parameters as follows:

![image](https://github.com/ClapeyronThermo/Clapeyron.jl/assets/130734903/53315241-edf2-43fd-936d-eab298039e9e)

I understand I am unable to make changes directly into your branch. I would like to model a couple of solids, I already have the molecular weights, melting enthalpies and temperatures.

May you kindly assist.

Thank you!

**Comment by pw0908** [comment]
Hi Kelethabetse,

The only thing you're missing from your code is that you need to specify the solid parameters for benzoic acid. You can do this within a csv file. Here is how I would set it up:
```julia
Clapeyron Database File
SolidHfus Like Parameters
species,Mw,Tm,Hfus
benzoic acid,122.1213,396.9,16990.
```
I saved it as `solid_params.csv`. From here, you can construct the model exactly as you have, only specifying this file as a new location for Clapeyron to search for parameters:
```julia
julia> model = CompositeModel([("ethanol",["CH3"=>1,"CH2OH"=>1]),("benzoic acid",["aCH"=>5,"aCCOOH"=>1])];fluid=SAFTgammaMie,solid=SolidHfus,solid_userlocations=["solid_params.csv"])
Composite Model:
 Gas Model: SAFTgammaMie{BasicIdeal, SAFTVRMie{BasicIdeal}}("ethanol", "benzoic acid")
 Liquid Model: SAFTgammaMie{BasicIdeal, SAFTVRMie{BasicIdeal}}("ethanol", "benzoic acid")
 Solid Model: SolidHfus("ethanol", "benzoic acid")

 Saturation Model: SAFTgammaMie{BasicIdeal, SAFTVRMie{BasicIdeal}}("ethanol", "benzoic acid")
```
From here you can obtain your solubilities:
```julia
julia> sle_solubility(model,1e5,298.15,[1.,1.];solute=["benzoic acid"])
2-element Vector{Float64}:
 0.4677379778031276
 0.5322620221968724
```

I will say that SAFT-gamma Mie doesn't seem to be doing that great a job here. You could consider using UNIFAC instead:
```julia
julia> model = CompositeModel([("ethanol",["CH3"=>1,"CH2"=>1,"OH (P)"=>1]),("benzoic acid",["ACH"=>5,"AC"=>1,"COOH"=>1])];liquid=UNIFAC,solid=SolidHfus,solid_userlocations=["like.csv"])
Composite Model:
 Gas Model: BasicIdeal()
 Liquid Model: UNIFAC{BasicIdeal}("ethanol", "benzoic acid")
 Solid Model: SolidHfus("ethanol", "benzoic acid")

 Saturation Model: LeeKeslerSat("ethanol", "benzoic acid")

julia> sle_solubility(model,1e5,275.,[1.,1.];solute=["benzoic acid"])
2-element Vector{Float64}:
 0.964191020149519
 0.03580897985048106
```
